nixos/modules/services/networking/wireguard.nix

```
@@ -316,12 +329,22 @@ let
           wg set ${name} private-key ${privKey} ${
             optionalString (values.listenPort != null) " listen-port ${toString values.listenPort}"}
 
+          ${optionalString (values.fwmark != null) ''
+            wg set ${name} fwmark "${toString values.fwmark}"
+            ip rule add not fwmark "${toString values.fwmark}" table "${toString values.table}"
+            ip rule add table "${values.table}" suppress_prefixlength 0
```

I can't stress enough how I have no idea what I'm doing.

---

Fancy table stuff should be conditionalized on /0 in allowed-ips, not on fwmark being set, since there are many uses of fwmark.

---

```suggestion
            ip rule add table main suppress_prefixlength 0
```

That line should be `main` - we want to suppress the default route in the main routing table.

This seems to also be missing IPv6 support.


I agree with @zx2c4 this shouldn't be conditionalized on fwmark being set, but we probably shouldn't conditionalize on "allowed-ips contain a default route" either.

Users might set `allowedIPsAsRoutes` to false, configure peers with `::/0`, `0.0.0.0/0`, and add routes manually, or have a routing daemon doing it - in that case, we don't really have control about whether a default route is inserted.

Still, we should provide an option to enable the "fancy table stuff" and configure the routing table name and fwmark number being used.